tests/ui/offset-of/offset-of-dst-field.rs

```diff
@@ -26,8 +26,30 @@ struct Gamma {
     z: Extern,
 }
 
+struct Delta<T: ?Sized> {
+    x: u8,
+    y: u16,
+    z: T,
+}
+
 fn main() {
     offset_of!(Alpha, z); //~ ERROR the size for values of type
     offset_of!(Beta, z); //~ ERROR the size for values of type
     offset_of!(Gamma, z); //~ ERROR the size for values of type
 }
+
+fn delta() {
+    offset_of!(Delta<u8>, z); // compiles fine
+    offset_of!(Delta<Alpha>, z); //~ ERROR the size for values of type
+    offset_of!(Delta<Extern>, z); //~ ERROR the size for values of type
+    offset_of!(Delta<dyn Trait>, z); //~ ERROR the size for values of type
+    offset_of!(Delta<Box<dyn Trait>>, z); // compiles fine
```

Comment:
FTR this might not actually ensure that the line compiles fine. I mean it does compile fine, but there is deduplication active for the "unsized" error, which means that if some type causes the error multiple times inside the same function, the error is only emitted once. That's why I moved the delta related stuff into its own function.

So maybe the best would be to move the `Box<dyn Trait>` stuff into its own function.